controllers/main.js

```
@@ -22,20 +22,42 @@ const getUrl = async (req, res) => {
     }
     res.json(url);
   } catch (error) {
-    res.status(HTTP_ERRORS.INTERNAL_SERVER_ERROR).json({ error: 'Something went wrong. Try Again' });
+    res
+      .status(HTTP_ERRORS.INTERNAL_SERVER_ERROR)
+      .json({ error: 'Something went wrong. Try Again' });
   }
 };
 
 const editUrl = async (req, res) => {
-  const { originalUrl, customName, shortUrl } = req.body;
+  const { originalUrl, customName } = req.body;
   const { id } = req.params;
   const filter = { _id: id };
-  const url = await Url.findOneAndUpdate(
-    filter,
-    { originalUrl, customName, shortUrl },
-    { new: true }
-  );
-  res.json(url);
+  let shortUrls;
+
+  try {
+    if (customName) {
+      const customNameWithDashes = customName.split(' ').join('-');
+      shortUrls = `${BASEURL}/${customNameWithDashes}`;
+    }
+
+    const url = await Url.findOneAndUpdate(
```

The first thing you should do before updating is find that the route parameter id is in the database.

When I send an id that does not exist. e.g: http://localhost:3000/api/v1/urls/6531dd08a605e11f97970359[PUT], it throws a server error. The endpoint should throw a not found error.

---

done on 3a08efb